Sources/NIOExtras/PCAPRingBuffer.swift

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+// MARK: NIOPCAPRingBuffer
+/// Storage for the most recent set of packets captured subject to constraints.
+/// Use `addFragment` as the sink to a `NIOWritePCAPHandler` and call `emitPCAP`
+/// when you wish to get the recorded data.
+public class NIOPCAPRingBuffer {
+    private var pcapFragments: CircularBuffer<ByteBuffer>
+    private var pcapCurrentBytes: size_t
+    private let maximumFragments: UInt
+    private let maximumBytes: size_t
+
+    /// Initialise the buffer, setting constraints.
+    /// Parameters:
+    ///    - maximumFragments: The maximum number of pcap fragments to store.
+    ///    - maximumBytes:  The maximum number of bytes to store - note, data written may exceed this by the header size.
```

Comment:
Do we want to support both conditions at once? It feels like limiting by `maximumFragments` OR `maximumBytes` would be more useful.

Reply:
Yeah, I think that would make the API clearer..